app/parsing_scripts/git_update.py

```diff
@@ -6,88 +6,83 @@
+                                stdout=subprocess.PIPE).stdout.decode('utf-8').strip()
+        if remote == MACPORTS_PORTS_URL:
+            print("The {} repo is healthy.".format(MACPORTS_PORTS))
+        else:
+            raise OSError
+
+        # If old_hash is not provided by the user
+        if old_hash is False:
+            # If the database has old commit, use it for old_hash
+            old_hash_object = LastPortIndexUpdate.objects.all().first()
+            if old_hash_object is None:
+                # If database is empty, use the first commit
+                old_hash = subprocess.run([GIT, 'rev-list', 'HEAD', '|', 'tail', '-n', '1'],
+                                          stdout=subprocess.PIPE).stdout.decode('utf-8')
             else:
-                raise OSError
-
-            # If old_hash is not provided by the user
-            if old_hash is False:
-                try:
-                    # Try to fetch the most recent hash from database
-                    old_hash_object = Commit.objects.all().order_by('-updated_at').first()
-                    old_hash = old_hash_object.hash
-                except AttributeError:
-                    # If database is empty, use the current HEAD
-                    old_hash = subprocess.run(['git', 'rev-parse', 'HEAD'], stdout=subprocess.PIPE).stdout.decode(
-                            'utf-8')
-
-            # Pull from macports-ports
-            subprocess.call(['git', 'pull'])
-
-            # Get new hash if not provided
-            if new_hash is False:
-                new_hash = subprocess.run(['git', 'rev-parse', 'HEAD'], stdout=subprocess.PIPE).stdout.decode('utf-8')
-
-            # Generate the range of commits to find updated paths
-            range = str(old_hash).strip() + "^.." + str(new_hash).strip()
-
-            changed_paths = subprocess.run(['git', 'diff', '--name-only', range], stdout=subprocess.PIPE).stdout.decode(
-                'utf-8')
-            s = io.StringIO(changed_paths)
-            updated_ports = []
-            for line in s:
-                portname = line.split('/')[1]
-                if portname not in updated_ports:
-                    updated_ports.append(portname)
-
-            # Add the new hash to the database
-            Commit.objects.create(hash=new_hash)
-            os.chdir(BASE_DIR)
-            return updated_ports
-
-        except OSError:
-            os.chdir(root)
-            print("{} repository has some error".format(REPO))
-            print("Cleaning current tree and cloning new repo.")
-            shutil.rmtree(REPO)
-            clone_repo()
-            return get_list_of_changed_ports(new_hash, old_hash, root)
-    else:
-        print('{} directory not found. Cloning into'.format(REPO))
+                old_hash = old_hash_object.git_commit_hash
+
+        # Pull from macports-ports
+        subprocess.call([GIT, 'pull'])
```

Comment:
This would be ok for now, but in case something goes wrong, it is not super reliable.